### PR TITLE
Dont create a log entry on email login

### LINKED ISSUE #3038: Logging in with email generates errors in log
Hi!

I've seen on my Nextcloud-server that if a user logs in with their registered e-mail, then they are able to logging in, but it throws errors to the server log. I've seen this in all versions up to 10.0.2, and I'm still experiencing the issue. If I ask users to log in with username only, then there's no errors in the log anymore.

### Server configuration
**Operating system**: Ubuntu 16.04.1

**Web server:** Apache

**Database:** Mysql 14.14

**PHP version:** 7.0.8

**Nextcloud version:** 10.0.2

**Updated from an older Nextcloud/ownCloud or fresh install:** Clean install

**Where did you install Nextcloud from:** Downloaded zip file from webpage

**Log entries:**
Warning	core	Login failed: 'marius@domain' (Remote IP: '10.0.0.0')	2017-01-12T07:02:53+00:00	--
Warning	core	Login failed: 'rolf@domain' (Remote IP: '10.0.0.0')	2017-01-12T06:59:17+00:00	--
Warning	core	Login failed: 'rolf@domain' (Remote IP: '10.0.0.0')	2017-01-12T06:59:11+00:00	--
Warning	core	Login failed: 'rolf@domain' (Remote IP: '10.0.0.0')	2017-01-12T06:59:11+00:00	--
Warning	core	Login failed: 'rolf@domain' (Remote IP: '10.0.0.0')	2017-01-12T06:59:10+00:00	--
Warning	core	Login failed: 'rolf@domain' (Remote IP: '10.0.0.0')	2017-01-12T06:59:11+00:00	--
Warning	core	Login failed: 'rolf@domain' (Remote IP: '10.0.0.0')	2017-01-12T06:59:10+00:00	--

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #3038

### CODE REVIEW

**Review by icewind1991 — APPROVED** [review]

File: lib/private/User/Manager.php [hunk 1]
```diff
@@ -196,8 +196,6 @@ public function checkPassword($loginName, $password) {
 				}
 			}
 		}
-
-		\OC::$server->getLogger()->warning('Login failed: \''. $loginName .'\' (Remote IP: \''. \OC::$server->getRequest()->getRemoteAddress(). '\')', ['app' => 'core']);
```
Comment (ChristophWurst) [inline lib/private/User/Manager.php]: I'm not a big fan of removing this here. What about other code that uses this method, like for example the user session? Don't we want to log failures there too?

Comment (ChristophWurst) [inline lib/private/User/Manager.php]: I think failed client auth attempts are also unnoticed then

Comment (nickvergessen) [inline lib/private/User/Manager.php]: Well but other code that uses this wrongly creates a "failed login" log, although just a password confirmation failed (changing a password with a wrong old password e.g.)

Alternative would be to add a parameter or second method which does the same but without logging...

Comment (MorrisJobke) [inline lib/private/User/Manager.php]: How to carry on here? I would vote for a silent flag if that is all that is needed.

Comment (ChristophWurst) [inline lib/private/User/Manager.php]: How about we check which code paths use that method and add the logging there?

Comment (nickvergessen) [inline lib/private/User/Manager.php]: That is basically what I already did....

Comment (ChristophWurst) [inline lib/private/User/Manager.php]: I don't think so. E.g. there's https://github.com/nextcloud/server/blob/847c974e1c52177fae06daf8397f711148dfbb96/lib/private/User/Session.php#L467 and https://github.com/nextcloud/server/blob/847c974e1c52177fae06daf8397f711148dfbb96/lib/private/User/Session.php#L632

Comment (MorrisJobke) [inline lib/private/User/Manager.php]: @nickvergessen Any statement from you? ;)

Comment (nickvergessen) [inline lib/private/User/Manager.php]: Well feel free to add it where you think it's needed.

Comment (ChristophWurst) [inline lib/private/User/Manager.php]: I will have a look next week 😉

Comment (LukasReschke) [inline lib/private/User/Manager.php]: > I will have a look next week 😉

🏓

**Comment by ChristophWurst** [comment]
There are many places where we use ``checkPassword`` and most of them should definitely log failed login attempts. Duplicating the logic there feels wrong, e.g. because we might forget to log when we add another consumer of the interface or if it's used inside apps where we cannot check it at all.

bildschirmfoto von 2017-02-20 19-16-08

The two possible solutions I can think of are to adding a flag to actively prevent logging failed attempts (what @MorrisJobke suggested above) or adding another method, like `checkPasswordSilently`, move the logging over and call the original method inside of the new one.

**Comment by nickvergessen** [comment]
Okay, so let me do that.

**Comment by nickvergessen** [comment]
Hmpf, looked into that, and it's really not very easy. We have to pipe this method/parameter through the manager `IUserManager` and the backends `UserInterface`

**Comment by MorrisJobke** [comment]
@nickvergessen @ChristophWurst What to do here? Could we find a solution for this issue?

**Comment by nickvergessen** [comment]
Updated the PR one more time. This is the best I can come up with.

**Review by ChristophWurst — APPROVED** [review]
Looks good now, I like that approach 👍

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍 Code looks good

**Pull request merged**